python/cudf/cudf/core/index.py

```diff
@@ -249,9 +252,29 @@ def _num_rows(self):
     @_cudf_nvtx_annotate
     def _values(self):
         if len(self) > 0:
-            return column.arange(
+            ret = column.arange(
                 self._start, self._stop, self._step, dtype=self.dtype
             )
+            manager = get_global_manager()
```

Comment:
I'm concerned about knowledge of the spilling manager leaking into other cuDF types like `RangeIndex`. Is there any way we can do this totally outside of `RangeIndex`?

Reply:
Introducing a `cached_property_delete_column_when_spilled` instread